modules/nodepool/main.tf

```diff
@@ -17,15 +17,18 @@ resource "aws_launch_template" "this" {
   user_data              = var.userdata
   vpc_security_group_ids = concat([aws_security_group.this.id], var.vpc_security_group_ids)
 
-  block_device_mappings {
-    device_name = lookup(var.block_device_mappings, "device_name", "/dev/sda1")
-    ebs {
-      volume_type           = lookup(var.block_device_mappings, "type", null)
-      volume_size           = lookup(var.block_device_mappings, "size", null)
-      iops                  = lookup(var.block_device_mappings, "iops", null)
-      kms_key_id            = lookup(var.block_device_mappings, "kms_key_id", null)
-      encrypted             = lookup(var.block_device_mappings, "encrypted", null)
-      delete_on_termination = lookup(var.block_device_mappings, "delete_on_termination", null)
+  dynamic "block_device_mappings" {
```

Comment:
I'm slightly worried about implicitly mixing the root block device (`/dev/sda1` by default) with any additional block devices.

Would it be possible to keep the existing `block_evice_mappings` block, but include a separate `dynamic "block_device_mappings" {}` that only loops over the `extra_block_device_mappings`?

Reply:
this would also imply that the default device name for `extra_block_device_mappings` would be something like `/dev/sdf` instead of `/dev/sda1`

Reply:
Ok, I believe we have made the changes you suggested. Please let me know if they are satisfactory.